### PR TITLE
V2: Args for data splitting

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
This PR just connects the splitting args to the code. It also comments out the splitting args that aren't ready yet. 

It also fixes a bug in common.py (`choices=RxnMode.keys(),`). Discussed in #497.'

Note: Support for splitting by molecular weight was added recently to v1, but has not been added to v2.

## Questions
Support for premade data splits could be added to this PR as well as support for saving smiles splits. These seemed more complicated so were left for later. One question is if they should be added to this PR instead.

Another question is if 'crossval' and 'cv' are the same splitting method. In v2 they show up as two options, which confused me because I thought 'cv' stood for 'crossval'. Currently 'crossval' isn't an option from the commandline.

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

**Review by davidegraff — APPROVED** [review]
👍 checks out

**Comment by KnathanM** [comment]
Rebased on v2/dev. Should still be ready to merge unless we want to expand its scope to implement the command line arguments related to splitting that it just comments out for now.

File: chemprop/cli/train.py [hunk 1]
```diff
@@ -381,11 +381,9 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
...
         "--split-type",
-        default="random",
-        choices=[
-            "random",
-            "scaffold",
-        ],  # 'scaffold_balanced', 'predetermined', 'crossval', 'cv', 'cv-no-test', 'index_predetermined', 'random_with_repeated_smiles'], # TODO: make data splitting CLI play nicely with astartes backend
+        type=uppercase,
+        default="RANDOM",
+        choices=list(SplitType.keys()),
         help="Method of splitting the data into train/val/test.",
```
Comment (davidegraff) [inline chemprop/cli/train.py:387]:
```suggestion
        help="Method of splitting the data into train/val/test (case insensitive).",
```

File: chemprop/cli/common.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def add_common_args(parser: ArgumentParser) -> ArgumentParser:
         "--reaction-mode",
         type=uppercase,
         default="REAC_DIFF",
-        choices=RxnMode.keys(),
+        choices=list(RxnMode.keys()),
         help="""Choices for construction of atom and bond features for reactions
```
Comment (davidegraff) [inline chemprop/cli/common.py:70]:
```suggestion
        help="""Choices for construction of atom and bond features for reactions (case insensitive):
```

**Review by davidegraff — DISMISSED** [review]
just small nits on the help message. Good to go from me even without these changes.

**Comment by KnathanM** [comment]
The checks were failing because some `import` bugs from previous PRs. I felt those fixes belonged in different PRs. So this should be good to merge after #578 and #579.

File: chemprop/cli/train.py (hunk 1 above)
Comment (KnathanM) [reply chemprop/cli/train.py]: added

**Review by kevingreenman — APPROVED** [review]

**Pull request merged**